client/src/components/SideBar/GroupSetting/index.tsx

```diff
@@ -1,29 +1,46 @@
-import React, { useState } from 'react';
-import { useSelectedGroup } from '@hooks/index';
+import React, { useEffect, useState } from 'react';
+import { useUserdata, useSelectedGroup } from '@hooks/index';
 import { ModalController } from '@customTypes/modal';
 import { GroupDeleteIcon, GroupInviteIcon } from '../../common/Icon';
 import GroupInviteModal from '../../Modal/GroupInvite';
+import GroupDeleteModal from '../../Modal/GroupDelete';
 import { GroupSettingWrapper } from './style';
 
 function GroupSetting() {
   const selectedGroup = useSelectedGroup();
+  const { userdata } = useUserdata();
+  const [selectedModal, setSelectedModal] = useState('');
+  const [isLeader, setIsLeader] = useState(false);
 
-  const [showModal, setShowModal] = useState(false);
-  const modalController: ModalController = {
-    hide: () => setShowModal(false),
-    show: () => setShowModal(true),
+  useEffect(() => {
```

Comment:
```suggestion
  useLayoutEffect(() => {
```
계속 useState를 사용할꺼라면 여기를 useLayoutEffect로 바꾸면 안깜빡일것!